doc/ml_features.rst

```diff
@@ -0,0 +1,337 @@
+    from smartredis import Client
+
+    # simple MNIST in PyTorch
+    class Net(nn.Module):
+        def __init__(self):
+            super(Net, self).__init__()
+            self.conv1 = nn.Conv2d(1, 32, 3, 1)
+            self.conv2 = nn.Conv2d(32, 64, 3, 1)
+            self.dropout1 = nn.Dropout(0.25)
+            self.dropout2 = nn.Dropout(0.5)
+            self.fc1 = nn.Linear(9216, 128)
+            self.fc2 = nn.Linear(128, 10)
+
+        def forward(self, x):
+            x = self.conv1(x)
+            x = F.relu(x)
+            x = self.conv2(x)
+            x = F.relu(x)
+            x = F.max_pool2d(x, 2)
+            x = self.dropout1(x)
+            x = torch.flatten(x, 1)
+            x = self.fc1(x)
+            x = F.relu(x)
+            x = self.dropout2(x)
+            x = self.fc2(x)
+            output = F.log_softmax(x, dim=1)
+            return output
+
+We can then create the Neural Network, jit-trace it and upload it
+to the DB. Note that we are storing the serialized model in a ``BytesIO``
+object, which means that we are keeping it in memory and not storing
+it on the file system. For this reason, we need to call SmartRedis's
+``Client.set_model()`` method.
+
+
+.. code-block:: python
+
+    n = Net()
+    example_forward_input = torch.rand(20, 1, 28, 28)
+    module = torch.jit.trace(n, example_forward_input)
+    model_buffer = io.BytesIO()
+    torch.jit.save(module, model_buffer)
+    net = model_buffer.getvalue()
+
+    # connect a client to the database
+    client = Client(cluster=False)
+
+    # 20 samples of "image" data
+    client.set_model("cnn", net, "TORCH", device="CPU")
+    client.put_tensor("input", example_forward_input.numpy())
+    client.run_model("cnn", inputs=["input"], outputs=["output"])
+    output = client.get_tensor("output")
+
+The model can be serialized and stored on disk by replacing the
+``BytesIO`` object by a string representing the file name. It can
+then be uploaded to the DB by calling ``set_model_from_file`` instead
+of ``set_model``.
+
+.. code-block:: python
+
+    n = Net()
+    example_forward_input = torch.rand(20, 1, 28, 28)
+    module = torch.jit.trace(n, example_forward_input)
+    torch.jit.save(module, "traced_model.pt")
+
+    # connect a client to the database
+    client = Client(cluster=False)
+
+    # 20 samples of "image" data
+    client.set_model_from_file("cnn", "traced_model.pt", "TORCH", device="CPU")
+    client.put_tensor("input", example_forward_input.numpy())
+    client.run_model("cnn", inputs=["input"], outputs=["output"])
+    output = client.get_tensor("output")
+
+
+TorchScript Functions
+---------------------
+Instead of Neural Networks, or, in general, Machine Learning models, it is
+possible to upload to the DB (collections of) functions which can be used e.g.
+to perform pre- or post-processing operations on tensors stored on the DB.
+
+Since the functions are going to be stored as TorchScript modules, they
+- need to be jit-traceable
```

Comment:
Just incase: for rst, if I did not place a space between a sentence and a list, it just added on to the sentence.